src/maidsafe/common/rsa.cc

```diff
@@ -162,6 +162,28 @@ Signature Sign(const PlainText& data, const PrivateKey& private_key) {
   return Signature(signature);
 }
 
+std::vector<byte> Sign(const std::vector<byte>& data, const PrivateKey& private_key) {
+  if (!private_key.Validate(crypto::random_number_generator(), 0)) {
```

Comment:
I think almost all of this is duplicated from the other `Sign` function.  It looks like we could probably move this to a function in unnamed namespace and have each overload call this?